File: 通讯协议/ProtoGen/output/handlers/csharp_all/GuildBossHandler.cs
/*-----------------------------------------------
 *本文件由代码生成器自动生成，
 *千万不要修改本文件的任何代码，
 *修改的的任何代码都会被覆盖掉！
 --------------------------------------------------*/

using System.IO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Pomelo.DotNetClient;

namespace pomelo.area
{
	public class GuildBossHandler
	{
		private PomeloClient _socket;

		public EnterGuildBossAreaResponse lastEnterGuildBossAreaResponse { get { return _socket.GetLastResponse<EnterGuildBossAreaResponse>(); } }
		public GetGuildBossInfoResponse lastGetGuildBossInfoResponse { get { return _socket.GetLastResponse<GetGuildBossInfoResponse>(); } }
		public GuildBossInspireResponse lastGuildBossInspireResponse { get { return _socket.GetLastResponse<GuildBossInspireResponse>(); } }
		static GuildBossHandler()
		{
			EventTypes.RegistPushType("area.guildBossPush.onHurtRankChangePush", typeof(OnHurtRankChangePush));
			EventTypes.RegistPushType("area.guildBossPush.onInspireSingleChangePush", typeof(OnInspireSingleChangePush));
			EventTypes.RegistPushType("area.guildBossPush.onInspireGuildChangePush", typeof(OnInspireGuildChangePush));
			EventTypes.RegistPushType("area.guildBossPush.onGuildBossOverPush", typeof(OnGuildBossOverPush));
			EventTypes.RegistPushType("area.guildBossPush.onQuitGuildBossPush", typeof(OnQuitGuildBossPush));
			EventTypes.RegistRequestType("area.guildBossHandler.enterGuildBossAreaRequest", typeof(EnterGuildBossAreaRequest), typeof(EnterGuildBossAreaResponse));
			EventTypes.RegistRequestType("area.guildBossHandler.getGuildBossInfoRequest", typeof(GetGuildBossInfoRequest), typeof(GetGuildBossInfoResponse));
			EventTypes.RegistRequestType("area.guildBossHandler.guildBossInspireRequest", typeof(GuildBossInspireRequest), typeof(GuildBossInspireResponse));
		}
		public GuildBossHandler(PomeloClient socket)
		{
			this._socket = socket;

		}
		public void enterGuildBossAreaRequest(Action<PomeloException,EnterGuildBossAreaResponse> cb,object option = null)
		{
			var request = new EnterGuildBossAreaRequest();
			_socket.request<EnterGuildBossAreaResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as EnterGuildBossAreaResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void getGuildBossInfoRequest(Action<PomeloException,GetGuildBossInfoResponse> cb,object option = null)
		{
			var request = new GetGuildBossInfoRequest();
			_socket.request<GetGuildBossInfoResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as GetGuildBossInfoResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void guildBossInspireRequest(int c2s_index,Action<PomeloException,GuildBossInspireResponse> cb,object option = null)
		{
			var request = new GuildBossInspireRequest();
			request.c2s_index= c2s_index;
			_socket.request<GuildBossInspireResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as GuildBossInspireResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void onHurtRankChangePush(Action<OnHurtRankChangePush> cb)
		{
			_socket.listen_once<OnHurtRankChangePush>(cb);
		}
		public void onInspireSingleChangePush(Action<OnInspireSingleChangePush> cb)
		{
			_socket.listen_once<OnInspireSingleChangePush>(cb);
		}
		public void onInspireGuildChangePush(Action<OnInspireGuildChangePush> cb)
		{
			_socket.listen_once<OnInspireGuildChangePush>(cb);
		}
		public void onGuildBossOverPush(Action<OnGuildBossOverPush> cb)
		{
			_socket.listen_once<OnGuildBossOverPush>(cb);
		}
		public void onQuitGuildBossPush(Action<OnQuitGuildBossPush> cb)
		{
			_socket.listen_once<OnQuitGuildBossPush>(cb);
		}

	}


}

File: 通讯协议/ProtoGen/output/handlers/lua/GuildBossHandler.lua
-------------------------------------------------
-- 本文件由代码生成器自动生成，
-- 千万不要修改本文件的任何代码，
-- 修改的的任何代码都会被覆盖掉！
----------------------------------------------------

local Socket = require "Xmds.Pomelo.LuaGameSocket"
require "base64"
require "guildBossHandler_pb"

--------------------------------Pomelo-------------------------------
Pomelo = Pomelo or {}

--------------------------------GuildBossHandler-------------------------------
Pomelo.GuildBossHandler = {}

local function enterGuildBossAreaRequestEncoder(msg)
	local input = guildBossHandler_pb.EnterGuildBossAreaRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function enterGuildBossAreaRequestDecoder(stream)
	local res = guildBossHandler_pb.EnterGuildBossAreaResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.GuildBossHandler.enterGuildBossAreaRequest(cb,option)
	local input = nil
	Socket.OnRequestStart("area.guildBossHandler.enterGuildBossAreaRequest", option)
	Socket.Request("area.guildBossHandler.enterGuildBossAreaRequest", input, function(res)
		if(res.s2c_code == 200) then
			Pomelo.GuildBossHandler.lastEnterGuildBossAreaResponse = res
			Socket.OnRequestEnd("area.guildBossHandler.enterGuildBossAreaRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.guildBossHandler.enterGuildBossAreaRequest decode error!!"
			end
			Socket.OnRequestEnd("area.guildBossHandler.enterGuildBossAreaRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, enterGuildBossAreaRequestEncoder, enterGuildBossAreaRequestDecoder)
end


local function getGuildBossInfoRequestEncoder(msg)
	local input = guildBossHandler_pb.GetGuildBossInfoRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function getGuildBossInfoRequestDecoder(stream)
	local res = guildBossHandler_pb.GetGuildBossInfoResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.GuildBossHandler.getGuildBossInfoRequest(cb,option)
	local input = nil
	Socket.OnRequestStart("area.guildBossHandler.getGuildBossInfoRequest", option)
	Socket.Request("area.guildBossHandler.getGuildBossInfoRequest", input, function(res)
		if(res.s2c_code == 200) then
			Pomelo.GuildBossHandler.lastGetGuildBossInfoResponse = res
			Socket.OnRequestEnd("area.guildBossHandler.getGuildBossInfoRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.guildBossHandler.getGuildBossInfoRequest decode error!!"
			end
			Socket.OnRequestEnd("area.guildBossHandler.getGuildBossInfoRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, getGuildBossInfoRequestEncoder, getGuildBossInfoRequestDecoder)
end


local function guildBossInspireRequestEncoder(msg)
	local input = guildBossHandler_pb.GuildBossInspireRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function guildBossInspireRequestDecoder(stream)
	local res = guildBossHandler_pb.GuildBossInspireResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.GuildBossHandler.guildBossInspireRequest(c2s_index,cb,option)
	local msg = {}
	msg.c2s_index = c2s_index
	Socket.OnRequestStart("area.guildBossHandler.guildBossInspireRequest", option)
	Socket.Request("area.guildBossHandler.guildBossInspireRequest", msg, function(res)
		if(res.s2c_code == 200) then
			Pomelo.GuildBossHandler.lastGuildBossInspireResponse = res
			Socket.OnRequestEnd("area.guildBossHandler.guildBossInspireRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] area.guildBossHandler.guildBossInspireRequest decode error!!"
			end
			Socket.OnRequestEnd("area.guildBossHandler.guildBossInspireRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, guildBossInspireRequestEncoder, guildBossInspireRequestDecoder)
end


local function onHurtRankChangePushDecoder(stream)
	local res = guildBossHandler_pb.OnHurtRankChangePush()
	res:ParseFromString(stream)
	return res
end

function Pomelo.GuildBossHandler.onHurtRankChangePush(cb)
	Socket.On("area.guildBossPush.onHurtRankChangePush", function(res) 
		Pomelo.GuildBossHandler.lastOnHurtRankChangePush = res
		cb(nil,res) 
	end, onHurtRankChangePushDecoder) 
end


local function onInspireSingleChangePushDecoder(stream)
	local res = guildBossHandler_pb.OnInspireSingleChangePush()
	res:ParseFromString(stream)
	return res
end

function Pomelo.GuildBossHandler.onInspireSingleChangePush(cb)
	Socket.On("area.guildBossPush.onInspireSingleChangePush", function(res) 
		Pomelo.GuildBossHandler.lastOnInspireSingleChangePush = res
		cb(nil,res) 
	end, onInspireSingleChangePushDecoder) 
end


local function onInspireGuildChangePushDecoder(stream)
	local res = guildBossHandler_pb.OnInspireGuildChangePush()
	res:ParseFromString(stream)
	return res
end

function Pomelo.GuildBossHandler.onInspireGuildChangePush(cb)
	Socket.On("area.guildBossPush.onInspireGuildChangePush", function(res) 
		Pomelo.GuildBossHandler.lastOnInspireGuildChangePush = res
		cb(nil,res) 
	end, onInspireGuildChangePushDecoder) 
end


local function onGuildBossOverPushDecoder(stream)
	local res = guildBossHandler_pb.OnGuildBossOverPush()
	res:ParseFromString(stream)
	return res
end

function Pomelo.GuildBossHandler.onGuildBossOverPush(cb)
	Socket.On("area.guildBossPush.onGuildBossOverPush", function(res) 
		Pomelo.GuildBossHandler.lastOnGuildBossOverPush = res
		cb(nil,res) 
	end, onGuildBossOverPushDecoder) 
end


local function onQuitGuildBossPushDecoder(stream)
	local res = guildBossHandler_pb.OnQuitGuildBossPush()
	res:ParseFromString(stream)
	return res
end

function Pomelo.GuildBossHandler.onQuitGuildBossPush(cb)
	Socket.On("area.guildBossPush.onQuitGuildBossPush", function(res) 
		Pomelo.GuildBossHandler.lastOnQuitGuildBossPush = res
		cb(nil,res) 
	end, onQuitGuildBossPushDecoder) 
end




--------------------------------Pomelo-------------------------------
return Pomelo
